[WIP] Refactor building part of hack/release.sh into generate-yamls.sh

tcnghia opened the pull request:
<!--
Request Prow to automatically lint any go code in this PR:

/lint
-->

## Proposed Changes

  * Refactor building part of hack/release.sh into separate script.  This allows us to change this part freely in different branches, while sharing the publishing & tagging logic between the branches

**Release Note**
<!-- Enter your extended release note in the below block. If the PR requires
additional action from users switching to the new release, include the string
"action required". If no release note is required, write "NONE". -->
```release-note
```

tcnghia commented:
/assign @adrcunha

tcnghia commented:
/test pull-knative-serving-integration-tests
--

tcnghia commented:
/test pull-knative-serving-integration-tests

tcnghia commented:
/assign @jessiezcc

jessiezcc commented:
can you give an example where we will need to have different logics of generating release YAML per branch?

tcnghia commented:
An example is the current state of our `release-0.1` branch and `HEAD`: some inputs YAMLs (that ko used) in `release-0.1` branches are moved, removed, renamed, or consolidated into different YAMLs in HEAD.  

This is the ko command in release-0.1

```
ko resolve -R -f config/monitoring/100-common \
    -f third_party/config/monitoring/common/istio \
    -f third_party/config/monitoring/common/kubernetes/kube-state-metrics \
    -f third_party/config/monitoring/common/prometheus-operator \
    -f config/monitoring/150-elasticsearch/100-scaling-configmap.yaml \
    -f config/monitoring/200-common/100-fluentd.yaml \
    -f config/monitoring/200-common/100-grafana-dash-knative-efficiency.yaml \
    -f config/monitoring/200-common/100-grafana-dash-knative.yaml \
    -f config/monitoring/200-common/100-grafana.yaml \
    -f config/monitoring/200-common/100-istio.yaml \
    -f config/monitoring/200-common/200-prometheus-exporter \
    -f config/monitoring/200-common/300-prometheus \
    -f config/monitoring/200-common/100-istio.yaml >> ${LITE_YAML}
```

vs at HEAD

```
echo "Copying Build release"
cp "${REPO_ROOT_DIR}/third_party/config/build/release.yaml" "${BUILD_YAML}"

echo "Building Knative Serving"
ko resolve ${KO_FLAGS} -f config/ > "${SERVING_YAML}"

# NO_MON is just build and serving
cp "${BUILD_YAML}" "${NO_MON_YAML}"
echo "---" >> "${NO_MON_YAML}"
cat "${SERVING_YAML}" >> "${NO_MON_YAML}"
echo "---" >> "${NO_MON_YAML}"
# LITE is NO_MON plus "lean" monitoring
cp "${NO_MON_YAML}" "${LITE_YAML}"
echo "---" >> "${LITE_YAML}"
cat "${MONITORING_METRIC_PROMETHEUS_YAML}" >> "${LITE_YAML}"
echo "---" >> "${LITE_YAML}"
```

tcnghia commented:
@jessiezcc I updated the PR description to explain that we will need to follow-up with the same refactoring in `release-0.1` branch.

jessiezcc commented:
/lgtm
/approve
